eZ/Bundle/EzPublishCoreBundle/Matcher/ServiceAwareMatcherFactory.php

```diff
@@ -38,10 +38,10 @@ public function __construct(
      */
     protected function getMatcher($matcherIdentifier)
     {
-        if (strpos($matcherIdentifier, '@') === 0) {
-            return $this->viewMatcherRegistry->getMatcher(substr($matcherIdentifier, 1));
+        try {
+            return $this->viewMatcherRegistry->getMatcher($matcherIdentifier);
```

Comment:
Matchers should actually be rather services and fully utilize Dependency Injection. If you want short identifiers, those should be literals associated with a given matcher (obtained either via method fulfilling a contract or as an attribute of tagged service (like `alias` in many other cases).

There's an underlying issue in [that portion of code](https://github.com/ezsystems/ezpublish-kernel/blob/v8.0.0-beta3/eZ/Publish/Core/MVC/Symfony/SiteAccess/MatcherBuilder.php#L29-L42) and [that](https://github.com/ezsystems/ezpublish-kernel/blob/dc7e85bb489e5e9b0c188701933da4b4929dca8f/eZ/Publish/Core/MVC/Symfony/Matcher/ClassNameMatcherFactory.php#L73-L95). Those should also utilize DI to get services.

In other words, the change could go in, provided we:
1. Introduce `identifier` for matchers.
2. Get rid of the exception-driven logic here (should always throw).
3. Drop dynamic namespace-based instantiation of matchers and register all of them as services.
4. Move a responsibility of getting matchers entirely to the registry (3. should solve that).

Reply:
Nothing to oppose to that.

The most important aspects for me, by order of priority:
- provide a clear, consistent syntax for _using_ matchers in config, independently of them being core or 3rd party, classes or services
- respect BC for those, as there are implementations around (mostly Netgen's, including Tags)

That aside, it doesn't bother me to make them all services. I can't recall the reasons why we chose to implement them that way. Maybe @lolautruche could shed some light ?